### ISSUE TITLE
Spelling checker

### ISSUE THREAD

**Issue by SparksJ2** [sorentycho/blackmirror#8]
Firstly, it's great that BM has a spelling cheker now. ;)

The current spelling checker, when turned on (ooer) only starts underlining incorrect words after editing a line, and then after that, only checks lines when you move the cursor to them. Expected behaviour would normally be to highlight all misspelled words on the document as soon as you activate it.

Also 'Add to dictionary' would be nice to have.

(Excuse me if I'm wrong (I don't javascript), but isn't electron basically chrome, and I thought chrome has a reasonably good spelling checker built in?)

**Comment by sorentycho** [comment]
Blocked on https://github.com/electron-userland/electron-spellchecker/issues/99

**Comment by sorentycho** [comment]
Partially fixed: I've implemented "add to dictionary".

**Comment by sorentycho** [comment]
> Excuse me if I'm wrong (I don't javascript), but isn't electron basically chrome, and I thought chrome has a reasonably good spelling checker built in?

Yes, but Electron is built on libchromiumcontent, which is *almost* Chrome, and so doesn't have any built-in spellchecker.

**Comment by SparksJ2** [comment]
Oh, how annoying.

While I'm asking for features, would it be difficult to add a country select? It's minor, but I prefer UK English.

**Comment by SparksJ2** [comment]
I tried 'add to dictionary' several times and couldn't get the red underline to go away, even after saving and reloading again.

**Comment by sorentycho** [comment]
Locale support is actually on the roadmap for spellchecking, via language tag at the top of `spelling.dict`.

As to add to dictionary, do you see the words being added to `~/conversations/spelling.dict`? If so, I'm guessing this is just a subclass of the original bug.

**Comment by SparksJ2** [comment]
Once I created `spelling.dict` manually, it's adding words to the file. Seems to be working now.
